### ISSUE TITLE
it should be possible to assign custom cost calculator to Map when in-memory-format is OBJECT

### ISSUE THREAD

**Issue by dsukhoroslov** [hazelcast/hazelcast#4699]
If I understand it correctly, memory cost is not calculated when in-memory-format is not BINARY, thus eviction policy based on memory consumption can't be used at all. It would be benefitial to allow users asign their own memory-cost-calculator for such case via Config interfaces and in xml configuration.

Thanks, Denis.

**Comment by noctarius** [comment]
Very good idea, love it. There is a new community procedure called Hazelcast Incubator https://hazelcast.atlassian.net/wiki/display/COM/Hazelcast+Enhancement+Proposals. Would you like to suggest it?

**Comment by dsukhoroslov** [comment]
Hi Chris,

Just did a HEP, please have a look.

Denis.

**Comment by noctarius** [comment]
Yep already saw it, will come back on that as quickly as possible :)

**Comment by noctarius** [comment]
There is your open proposal https://hazelcast.atlassian.net/wiki/display/COM/Map+Entry+Cost+Calculator+SPI - will post it for internal discussion :) thanks for the submission!

**Comment by dsukhoroslov** [comment]
glad to help, thanks :)

**Comment by dsukhoroslov** [comment]
Hi @noctarius, any update on this? I see the EntryCostEstimator interface was added, which can be used to calc cost for entries in OBJECT format also. Just need an ability to specify its (or, its factory) implementation on IMap/ICache level.
